pydevd.py

```diff
@@ -957,11 +976,17 @@ def run(self, file, globals=None, locals=None, is_module=False, set_trace=True):
                 # The file being run must be in the pythonpath (even if it was not before)
                 sys.path.insert(0, os.path.split(file)[0])
 
-            self.prepare_to_run()
-
             while not self.ready_to_run:
                 time.sleep(0.1)  # busy wait until we receive run command
 
+            if self.break_on_caught_exceptions or (self.plugin and self.plugin.has_exception_breaks()) or self.signature_factory:
+                # disable frame evaluation if there are exception breakpoints with 'On raise' activation policy
```

Comment:
Yes, you're right. I've added checks for adding exception breakpoints during run.
But I didn't get your phrase about `trace_exception`. What do you mean here? Frame evaluation debugger can't handle exceptions.